packages/kit/src/runtime/client/client.js

```diff
@@ -140,10 +140,16 @@ function clear_onward_history(current_history_index, current_navigation_index) {
  * Returns a `Promise` that never resolves (to prevent any
  * subsequent work, e.g. history manipulation, from happening)
  * @param {URL} url
+ * @returns {Promise<any>}
  */
 function native_navigation(url) {
 	location.href = url.href;
-	return new Promise(() => {});
+	return new Promise((resolve) => {
+		// if we're still on the same page, it was probably a download
```

Comment:
this has me a bit uneasy. Are there any situations where this can be true but it is not a download?

Reply:
only if native_navigation is being used to navigate to the same page. However, it's currently only used when navigating to another page, clicking a link that has a different href than the current page

Reply:
It's tricky. Some downloads are only detectable once the response headers have been downloaded and `Content-Disposition: attachment` is present.